review_practice/react-app/src/containers/RoomList/index.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useContext } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+
+import * as Style from './style';
+
+import { SearchRoomContext } from 'pages/SearchRoom';
+
+import RoomItem from 'components/RoomItem';
+import Modal from 'components/Modal';
+import ReservationModalBody from './ReservationModalBody';
+
+export default () => {
+    const {
+        searchOption: { query, searchOptions },
+    } = useContext(SearchRoomContext);
+    const [reserveButtonState, setReserveButtonState] = useState({ isClicked: false, roomId: null });
+    const { loading, error, data } = useQuery(query, {
+        variables: { searchOptions: searchOptions },
+    });
+
+    if (loading) return <Style.Loading>Loading...</Style.Loading>;
+    if (error) alert('숙소를 불러오는데 실패했습니다');
+
+    const reserveButtonHandle = ({ roomId }) => {
+        const state = reserveButtonState.isClicked
+            ? { ...reserveButtonState, isClicked: false, roomId: null }
+            : { ...reserveButtonState, isClicked: true, roomId };
+        setReserveButtonState(state);
+    };
+
+    const getRoomById = roomId => {
+        return data.rooms.filter(({ room }) => room.id === roomId)[0].room;
```

Comment:
```suggestion
        return data.rooms.find(({ room }) => room.id === roomId).room;
```

하나를 찾는 함수이기 때문에 `Array.prototype.find` 로 대체할 수 있습니다